### PR TITLE
[SPARK-20284][CORE] Make {Des,S}erializationStream extend Closeable

### PR DESCRIPTION
**Pull request by superbobry**
## What changes were proposed in this pull request?

This PR allows to use `SerializationStream` and `DeserializationStream` in try-with-resources.

## How was this patch tested?

`core` unit tests.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/serializer/Serializer.scala [hunk 1]
```diff
@@ -125,7 +125,7 @@ abstract class SerializerInstance {
  * A stream for writing serialized objects.
  */
 @DeveloperApi
-abstract class SerializationStream {
+abstract class SerializationStream extends Closeable {
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/serializer/Serializer.scala]: Does the close() method need 'override' then?

Comment (superbobry) [inline core/src/main/scala/org/apache/spark/serializer/Serializer.scala]: Sure, added that.

**Review by srowen — APPROVED** [review]
My only concern would be whether this changes the classes in a way that makes them subtly not compatible at runtime. MiMa doesn't seem to think so though, and it's a developer API, and I also can't think of why it would change compatibility of bytecode (close() method already exists) So I think this is OK.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**